### PR TITLE
Improvements on Rivets implementation

### PR DESCRIPTION
**Pull request by mikeric**
This improves on the implementation originally submitted by @altmind in #1176.

I rewrote things to make use of components instead of a single all-encompassing app object. I also replaced the domain-specific binders with generic/reusable ones. The adjustments I made are just so that it showcases components a bit and how you might start at writing a larger application using Rivets.

### CODE REVIEW

File: examples/rivets/index.html [hunk 1]
```diff
@@ -0,0 +1,119 @@
+<!doctype html>
+
```
Comment (arthurvr) [inline examples/rivets/index.html]: Remove this line

**Comment by arthurvr** [comment]
Thanks @mikeric, and thanks @altmind, of course! Things that need to happen:

* [ ] Add it to [the home page](https://github.com/tastejs/todomvc/blob/master/index.html#L272), under labs. 
* [ ] Make sure the code passes JSHint
* [ ] Make sure the code passes JSCS.
* [ ] Add all files needed to run the app to the repo (and gitignore). Right now it has some 404s.
* [ ] Address all 22 failing tests.
* [ ] Squash commits

Would you be willing to work through these bullet points, @mikeric?

**Comment by mikeric** [comment]
@arthurvr Hey, sorry for all the extra pull-requests.. New one is over here (#1226) with commits squashed, and the above points addressed :)

**Comment by arthurvr** [comment]
Closing in favor of https://github.com/tastejs/todomvc/pull/1226.

**Pull request closed without merging**